### PR TITLE
now using synthetics

### PR DESCRIPTION
**Pull request by regmarcos**
Hi guys! This pr is just to show you the base project, but with a difference. I'll be using synthetics instead of data binding because we always use synthetics,so i made a little refactor.
pr1

### CODE REVIEW

File: app/src/main/java/com/example/android/codelabs/navigation/MainActivity.kt [hunk 1]
```diff
@@ -28,6 +28,7 @@ import androidx.navigation.NavController
 import androidx.navigation.fragment.NavHostFragment
 import androidx.navigation.ui.AppBarConfiguration
 import com.google.android.material.navigation.NavigationView
+import kotlinx.android.synthetic.main.navigation_activity.*
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/android/codelabs/navigation/MainActivity.kt]: Pls remove the wildcard .

File: app/src/main/java/com/example/android/codelabs/navigation/fragments/DeepLinkFragment.kt [hunk 2]
```diff
@@ -25,11 +25,11 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.Button
-import android.widget.EditText
-import android.widget.TextView
 import androidx.core.app.NotificationCompat
 import androidx.fragment.app.Fragment
 import androidx.navigation.fragment.findNavController
+import com.example.android.codelabs.navigation.R
+import kotlinx.android.synthetic.main.deeplink_fragment.*
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/android/codelabs/navigation/fragments/DeepLinkFragment.kt]: Pls remove the wilcard.

File: app/src/main/java/com/example/android/codelabs/navigation/fragments/DeepLinkFragment.kt [hunk 3]
```diff
@@ -45,15 +45,13 @@ class DeepLinkFragment : Fragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-
-        val textView = view.findViewById<TextView>(R.id.text)
-        textView.text = arguments?.getString("myarg")
+        text_deeplink.text = arguments?.getString("myarg")
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/android/codelabs/navigation/fragments/DeepLinkFragment.kt]: Pls, use a const or a TAG for this hardcode

File: app/src/main/res/layout/home_fragment.xml [hunk 4]
```diff
@@ -18,10 +18,10 @@
...
     android:layout_height="match_parent"
-    tools:context="com.example.android.codelabs.navigation.HomeFragment">
+    tools:context="com.example.android.codelabs.navigation.fragments.HomeFragment">
 
     <TextView
-        android:id="@+id/text"
+        android:id="@+id/text_home"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
         android:layout_marginTop="8dp"
```
Comment (mauroLonghi) [inline app/src/main/res/layout/home_fragment.xml]: Pls, remove the hardcode

File: app/src/main/java/com/example/android/codelabs/navigation/fragments/DeepLinkFragment.kt (hunk 2 above)
Comment (regmarcos) [reply app/src/main/java/com/example/android/codelabs/navigation/fragments/DeepLinkFragment.kt]: done

File: app/src/main/java/com/example/android/codelabs/navigation/MainActivity.kt (hunk 1 above)
Comment (regmarcos) [reply app/src/main/java/com/example/android/codelabs/navigation/MainActivity.kt]: done

File: app/src/main/java/com/example/android/codelabs/navigation/fragments/DeepLinkFragment.kt (hunk 3 above)
Comment (regmarcos) [reply app/src/main/java/com/example/android/codelabs/navigation/fragments/DeepLinkFragment.kt]: done

File: app/src/main/res/layout/home_fragment.xml (hunk 4 above)
Comment (regmarcos) [reply app/src/main/res/layout/home_fragment.xml]: done

File: app/src/main/java/com/example/android/codelabs/navigation/MainActivity.kt (hunk 1 above)
Comment (mauroLonghi) [reply app/src/main/java/com/example/android/codelabs/navigation/MainActivity.kt]: Sorry I continue saw this wildcard

**Review by mauroLonghi — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

File: app/src/main/java/com/example/android/codelabs/navigation/DeepLinkAppWidgetProvider.kt [hunk 5]
```diff
@@ -37,7 +37,7 @@ class DeepLinkAppWidgetProvider : AppWidgetProvider() {
         )
 
         val args = Bundle()
-        args.putString("myarg", "From Widget")
+        args.putString(MY_ARG, "From Widget")
         // TODO STEP 10 - construct and set a PendingIntent using DeepLinkBuilder
 //        val pendingIntent = NavDeepLinkBuilder(context)
```
Comment (SartiMau) [inline app/src/main/java/com/example/android/codelabs/navigation/DeepLinkAppWidgetProvider.kt:42]: is this code necessary?? if not, please remove it

File: app/src/main/res/navigation/mobile_navigation.xml [hunk 6]
```diff
@@ -67,14 +67,14 @@
     <!-- TODO STEP 4 Create a new navigation destination pointing to SettingsFragment -->
```
Comment (SartiMau) [inline app/src/main/res/navigation/mobile_navigation.xml:67]: is this necessary?

**Pull request merged**